src/main/java/edu/cornell/kfs/rass/batch/RassProjectDirectorConverter.java

```diff
@@ -1,74 +1,101 @@
-	}
+    protected PersonService personService;
+    protected RoleService roleService;
+    protected Class<? extends CGProjectDirector> projectDirectorImplementationClass;
+    protected String projectDirectorPrimaryIndicatorPropertyName;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object convert(Class<? extends PersistableBusinessObject> businessObjectClass, RassPropertyDefinition propertyMapping, Object propertyValue) {
+        List<RassXMLAwardPiCoPiEntry> rassAwardPiCoPiEntries = (List<RassXMLAwardPiCoPiEntry>) propertyValue;
+        List<CGProjectDirector> projectDirectors = new ArrayList<>();
+        if (rassAwardPiCoPiEntries != null && rassAwardPiCoPiEntries.size() > 0) {
+            for (RassXMLAwardPiCoPiEntry rassAwardPiCoPi : rassAwardPiCoPiEntries) {
+                CGProjectDirector projectDirector = createNewProjectDirectorInstance();
+                Person projectDirectorPerson = personService.getPersonByPrincipalName(rassAwardPiCoPi.getProjectDirectorPrincipalName());
+                if (ObjectUtils.isNull(projectDirectorPerson)) {
+                    throw new RuntimeException("Cannot find person with principal name \"" + rassAwardPiCoPi.getProjectDirectorPrincipalName() + "\"");
+                }
+
+                if (!doesPersonHaveProjectDirectRole(projectDirectorPerson)) {
+                    LOG.info("convert, " + projectDirectorPerson.getPrincipalName() + " needs to be added to the project director role");
+                    roleService.assignPrincipalToRole(projectDirectorPerson.getPrincipalId(), KFSConstants.CoreModuleNamespaces.KFS,
+                            KFSConstants.SysKimApiConstants.CONTRACTS_AND_GRANTS_PROJECT_DIRECTOR, new HashMap<String, String>());
+                }
+
+                projectDirector.setPrincipalId(projectDirectorPerson.getPrincipalId());
+                ObjectPropertyUtils.setPropertyValue(projectDirector, projectDirectorPrimaryIndicatorPropertyName,
+                        getNullSafePrimaryDirectorFlag(rassAwardPiCoPi));
+                projectDirectors.add(projectDirector);
+            }
+        }
+
+        return projectDirectors;
+    }
 
     protected CGProjectDirector createNewProjectDirectorInstance() {
         if (projectDirectorImplementationClass == null) {
             throw new IllegalStateException("Project director implementation class cannot be null");
         }
-        
+
         try {
             return projectDirectorImplementationClass.newInstance();
         } catch (IllegalAccessException | InstantiationException e) {
-            throw new RuntimeException("Unable to create project director of type "
-                    + projectDirectorImplementationClass.getName(), e);
+            throw new RuntimeException("Unable to create project director of type " + projectDirectorImplementationClass.getName(), e);
         }
     }
 
+    protected boolean doesPersonHaveProjectDirectRole(Person person) {
```

Comment:
should this be doesPersonHaveProjectDirectorRole instead of doesPersonHaveProjectDirectRole?
Director vs Direct

Reply:
OK, I can change